### PR TITLE
Api key

### PR DESCRIPTION
**Pull request by mitread**
Added functions to support VSAC interaction using API Key authentication.  Added deprecation warnings to the existing VSAC functions.  Added unit tests for all the API key function interactions. Updated the README to note which functions/environment variables to use, as well as a note about the authentication deprecation.

### CODE REVIEW

**Review by cmoesel — CHANGES_REQUESTED** [review]
This looks really good.  I did a manual test as well to confirm it is working.  I pushed up that test to this branch so you can try it to (and because I think it may be helpful in the future).

I also noted a few typos and made one suggestion about possibly moving the deprecation warning.

File: lib/CodeService.js [hunk 1]
```diff
@@ -84,10 +86,36 @@ class CodeService {
...
+    // First, filter out the value sets we already have
+    const filteredVSList = valueSetList.filter(vs => {
+      const result = this.findValueSet(vs.id, vs.version);
+      return typeof result === 'undefined';
+    });
+    // Now download from VSAC if necessary
+    if (filteredVSList.length == 0) {
+      return Promise.resolve();
+    } else if ( typeof umlsAPIKey === 'undefined' || umlsAPIKey == null) {
+      return Promise.reject('Failed to download value sets since UMLS_API_KEYis not set.');
```
[the remark this answers is not in the document]
Comment (mitread) [reply lib/CodeService.js]: Fixed!

File: test/test.js [hunk 2]
```diff
@@ -663,6 +854,29 @@ describe('CodeService', function() {
...
+
+        const vsList = [
+          {name: 'Systolic Blood Pressure', id: '2.16.840.1.113883.3.526.3.1032', version: '20170320'}
+        ];
+        return service.ensureValueSetsWithAPIKey(vsList, null)
+          .then(function() {
+            should.fail(0, 1, 'This code should never be executed');
+          })
+          .catch(function(error) {
+            error.should.eql('Failed to download value sets since UMLS_API_KEYis not set.');
```
[the remark this answers is not in the document]
Comment (mitread) [reply test/test.js]: Fixed!

File: lib/download-vsac.js [hunk 3]
```diff
@@ -6,7 +6,20 @@ const parseVSACXML = require('./parse-vsac');
...
+}
+
+
+/*
+ * @deprecated:  As of Jan 1 2021 VSAC will no longer accept accept username and password.
+ * Please use downloadFromVSACWithAPIKey instead.
+ */
 function downloadFromVSAC(username, password, input, output, vsDB={}, caching=true) {
+  //deprecation warning
+  vsacAuthWarning('downloadFromVSAC', 'downloadFromVSACWithAPIKey');
```
[the remark this answers is not in the document]
Comment (mitread) [reply lib/download-vsac.js]: I removed the console warnings in `download-vsac` but left the `@deprecated` notice in the comments above the appropriate methods.  I added console warnings to functions you mentioned in `CodeService`

**Review by cmoesel — APPROVED** [review]
Looks good and works well!  Thanks for this, @mitread!

**Pull request merged**